File: samples/msgext-northwind-inventory-action-csharp/NorthwindDB/Suppliers.cs
using NorthwindInventory.Models; 

namespace NorthwindInventory.NorthwindDB
{
    public static class SupplierService
    {
        private static readonly string TableNameSupplier = "Suppliers";

        public static async Task<string> CreateSupplierAsync(Supplier supplier, IConfiguration configuration)
        {
            var dbService = new DbService<Supplier>(false, configuration);

            var nextId = await dbService.GetNextIdAsync(TableNameSupplier);

            var newSupplier = new Supplier
            {
                ETag = "",
                PartitionKey = TableNameSupplier,
                RowKey = nextId.ToString(),
                Timestamp = DateTimeOffset.UtcNow,
                SupplierID = nextId,
                CompanyName = supplier.CompanyName,
                ContactName = supplier.ContactName,
                ContactTitle = supplier.ContactTitle,
                Address = supplier.Address,
                City = supplier.City,
                Region = supplier.Region,
                PostalCode = supplier.PostalCode,
                Country = supplier.Country,
                Phone = supplier.Phone,
                Fax = supplier.Fax,
                HomePage = supplier.HomePage
            };

            await dbService.CreateEntityAsync(TableNameSupplier, newSupplier.RowKey, newSupplier);

            return newSupplier.SupplierID.ToString();
        }
    }
}

File: samples/msgext-northwind-inventory-action-ts/src/northwindDB/suppliers.ts
import {
    TABLE_NAME, Supplier
} from './model';

import DbService, { DbSupplier } from './dbService';

export async function createSupplier(supplier: Supplier): Promise<string> {  
    const dbService = new DbService<DbSupplier>(false);
    const nextId = await dbService.getNextId(TABLE_NAME.SUPPLIER);
    const newSupplier: DbSupplier = {
        etag: "",
        partitionKey: TABLE_NAME.SUPPLIER,
        rowKey: nextId.toString(),
        timestamp: new Date(),
        SupplierID: nextId.toString(),
        CompanyName: supplier.CompanyName,
        ContactName: supplier.ContactName,
        ContactTitle: supplier.ContactTitle,
        Address:supplier.Address,
        City: supplier.City,
        Region: "",
        PostalCode: "",
        Country: "",
        Phone: "",
        Fax: "",
        HomePage: ""
    };
    await dbService.createEntity(TABLE_NAME.SUPPLIER, newSupplier.SupplierID, newSupplier);
    return newSupplier.SupplierID;
};
